cmd/soroban-cli/src/deploy.rs

```diff
@@ -112,36 +116,59 @@ pub enum Error {
         contract_id: String,
         error: FromHexError,
     },
+    #[error("cannot parse WASM hash {wasm_hash}: {error}")]
+    CannotParseWasmHash {
+        wasm_hash: String,
+        error: FromHexError,
+    },
     #[error("cannot parse secret key")]
     CannotParseSecretKey,
     #[error(transparent)]
     Rpc(#[from] rpc::Error),
+    #[error("either --wasm or --wasm-hash has to be provided")]
```

Comment:
I suspected that such option may exist, however I couldn't find any comprehensive doc on available clap options. Any pointers/name of the option would be appreciated.

Reply:
I think you can assign both parameters to the same arg group with `group = "somegroupname"`, and then make the arg group required.

Ref: https://docs.rs/clap/3.2.23/clap/builder/struct.Arg.html#method.group

Then I think we define the arg group on the command/parser/args struct, which you can also include the names of the args in instead of on the individual fields:

```rust
#[clap(group(
    clap::ArgGroup::new("somegroupname")
        .required(true)
        .args(&["wasm", "wasm-hash"]),
))]
```

Ref: https://github.com/clap-rs/clap/discussions/3899

Reply:
Thanks! Updated code to use the group.

Reply:
I just tried this at the command line, it looks beautiful in the usage:
```
USAGE:
    soroban deploy [OPTIONS] <--wasm <WASM>|--wasm-hash <WASM_HASH>>
```